spec/features/admin/presence_monitoring/index_spec.rb

```diff
@@ -19,4 +19,41 @@
       expect(page).to have_content(check.presence)
     end
   end
+
+  describe 'pagination' do
+    before(:each) do
+      @third_page_first = create(:slack_presence_check)
+      @second_page_last = create(:slack_presence_check)
+      @checks_two = create_list(:slack_presence_check, 48)
+      @second_page_first = create(:slack_presence_check)
+      @first_page_last = create(:slack_presence_check)
+      @checks_one = create_list(:slack_presence_check, 49)
+    end
+
+    it 'displays the slack presence checks in a paginated manner, with only 50 per page' do
+      visit '/admin/slack_presence_checks'
+
+      expect(page).to have_content(@first_page_last.student.name)
+      expect(page).to have_content(@first_page_last.student.slack_id)
+      expect(page).to_not have_content(@second_page_first.student.name)
+      expect(page).to_not have_content(@second_page_first.student.slack_id)
+      expect(page).to_not have_content(@second_page_last.student.name)
+      expect(page).to_not have_content(@second_page_last.student.slack_id)
+      expect(page).to_not have_content(@third_page_first.student.name)
+      expect(page).to_not have_content(@third_page_first.student.slack_id)
+    end
+
+    it 'has navigation buttons that move between the pages' do
+      visit '/admin/slack_presence_checks'
+
+      expect(page).to have_link('Next', href: '/admin/slack_presence_checks/page/2')
+      expect(page).to have_link('2', href: '/admin/slack_presence_checks/page/2')
+      expect(page).to have_link('3', href: '/admin/slack_presence_checks/page/3')
+      expect(page).to have_link('Last', href: '/admin/slack_presence_checks/page/3')
+      click_link('Next')
+      expect(current_path).to eq('/admin/slack_presence_checks/page/2')
+    end
```

Comment:
Your testing is looking really good so far! Let's change these to expect query params per my above comments. Let's also add an expectation for seeing the correct students on the page after clicking the next link. It also looks like the paginate method will add links for first/last, so let's test those as well.

Reply:
I added an additional test checking for the additional links, and the presence of the students' names on the appropriate page